File: Data Structures & Algorithms/binary-search/submission-3.java
class Solution {
    public int search(int[] nums, int target) {
        int low = 0;
        int high = nums.length - 1;
        int mid = (high+low)/2;

        while (nums[mid] != target) {
            if (target > nums[mid]) {
                low = mid + 1;
            }
            else {
                high = mid - 1;
            }
            mid = (low+high)/2;
            if (low > high) {
                return -1;
            }
        }
        return mid;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-3.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        def helper(index, current, total):
            #print(index, current, total)
            if total == target:
                res.append(current.copy())
                return
            if index >= len(nums) or total > target:
                return
            current.append(nums[index])
            helper(index, current, total + nums[index])
            current.pop()
            helper(index + 1, current, total)
        helper(0,[],0)
        return res